odilia/src/state.rs

```diff
@@ -37,18 +34,11 @@ pub struct ScreenReaderState {
 	pub cache: Arc<Cache>,
 }
 
-enum ConfigType {
-	CliOverride,
-	XDGConfigHome,
-	Etc,
-	CreateDefault,
-}
-
 impl ScreenReaderState {
 	#[tracing::instrument]
 	pub async fn new(
 		ssip: Sender<SSIPRequest>,
-		config_override: Option<&std::path::Path>,
+		config: ApplicationConfig,
```

Comment:
Taking ownership makes sense here. Good call!

Reply:
yep, it seemned fitting at the moment, and still is.

Services which need initialising before state does, such as logging, will be passed the slice of the config that concerns them, by immutable reference. Then, state will be initialized last, of course taking ownership of the config, to in turn hand it in the same fassion to the services it initialises, such as speech.